site/oauth2-examples-oauth0.md

```diff
@@ -0,0 +1,134 @@
+# Use https://auth0.com/ as OAuth 2.0 server
+
+We are going to test 3 OAuth flows:
+1. Access management ui via a browser
+2. Access management rest api
+3. Access AMQP protocol
+
+## Prerequisites to follow this guide
+
+* Have an account in https://auth0.com/.
+* Docker
+
+## Create rabbitmq API
+
+In OAuth0, resources are mapped to Application APIs. Once you have logged onto your account in https://auth0.com/, go to dashboard > Applications > APIs > Create an API.
+
+* Give it the name `rabbitmq`. The important thing here is the `identifier` which must have the name of the *resource_server_id* we configured in RabbitMQ. This `identifier` goes into the `audience` JWT field. In our case, it is called `rabbitmq`
+* Choose `RS256` as the signing algorithm
+* Enable **RBAC**
+* Enable **Add Permissions in the Access Token**
+
+### Configure permissions in rabbitmq API
+
+Edit the API we just created with the name `rabbitmq`. Go into Permissions and add the permissions (scope) this api can grant.
+We are going to add the following scopes:
+* `rabbitmq.read:*/*`
+* `rabbitmq.write:*/*`
+* `rabbitmq.configure:*/*`
+* `rabbitmq.tag:administrator`
+
+### Create an OAuth client for the Management UI
+
+By default, for every API we create, an *Application* gets created using the API's `identifier` as its name.
+An *Application* requests an **OAuth client**.
+
+Go to dashboard > Applications, and you should see your application listed. An application gives us a *client_id*, a *client_secret* and a http endpoint called *Domain* where to claim a token.
+
+## Create Application rabbitmq-management
+
+An application gives us the client-id and client-secret for the management ui to authenticate on behalf
+of the end user.
+
+In setting we choose:
+* Application type : `Single Page applications`
+* Token Endpoint Authentication Method:  `None`
+* Allowed Callback URLs: `http://localhost:15672/js/oidc-oauth/login-callback.html`
+* Allowed Web Origins: `http://localhost:15672`
+* Allowed Origins (CORS): `http://localhost:15672`
+
+
+## Create a new user
+
+You can use your current OAuth0 user to login to RabbitMq or create a dedicated user for that. Up to you.
+
+### Authorize rabbitmq-management application
+
+Go to "Authorized Applications", click on "Authorize" and select all the scopes.
+
```

Comment:
Again should the UI options to select here be bolded, they are bolded in other procedures: "Go to "Authorized Applications", click on "Authorize" and select all the scopes. "

Keep styling consistent in documentation where possible

GLOBAL comment above and below where applicable.